internal/entitlement/metered/balance.go

```diff
@@ -95,26 +95,27 @@ func (e *connector) GetEntitlementBalanceHistory(ctx context.Context, entitlemen
 	// TODO: we should guard against abuse, getting history is expensive
 
 	// validate that we're working with a metered entitlement
-	ent, err := e.entitlementRepo.GetEntitlement(ctx, entitlementID)
+	entRepoEntity, err := e.entitlementRepo.GetEntitlement(ctx, entitlementID)
 	if err != nil {
 		return nil, credit.GrantBurnDownHistory{}, err
 	}
-	_, err = ParseFromGenericEntitlement(ent)
-	if err != nil {
-		return nil, credit.GrantBurnDownHistory{}, err
+
+	if entRepoEntity == nil {
+		return nil, credit.GrantBurnDownHistory{}, &entitlement.NotFoundError{EntitlementID: entitlementID}
```

Comment:
can this actually happen?

Reply:
Nope, but I would prefer to be a bit defensive regardless in case somebody refactors the code.